components/director/internal/open_resource_discovery/client.go

```diff
@@ -173,24 +170,47 @@ func buildDocumentURL(docURL, baseURL string) (string, error) { // TODO: tests w
 	return baseURL + docURL, nil
 }
 
-func buildWellKnownEndpoint(u string) (string, error) {
-	parsedURL, err := url.ParseRequestURI(u)
+func calculateBaseURL(webhookURL string, config WellKnownConfig) (string, error) {
```

Comment:
can we improve the first part of the method to something like 
```
	if config.BaseURL != "" {
		return config.BaseURL, nil
	}

	parsedWebhookURL, err := url.ParseRequestURI(webhookURL)
	if err != nil {
		return "", errors.New("error while parsing input webhook url")
	}

	if strings.HasSuffix(parsedWebhookURL.Path, WellKnownEndpoint) {
		strippedPath := strings.ReplaceAll(parsedWebhookURL.Path, WellKnownEndpoint, "")
		parsedWebhookURL.Path = strippedPath
		return parsedWebhookURL.String(), nil
	}
```

Reply:
It is cleaner, yes.